runtime/src/accounts_db.rs

```diff
@@ -1397,8 +1430,57 @@ impl AccountsDB {
         hashes
     }
 
+    pub fn freeze_accounts(
+        &mut self,
+        ancestors: &HashMap<Slot, usize>,
+        account_pubkeys: &[Pubkey],
+    ) {
+        for account_pubkey in account_pubkeys {
+            if let Some((account, _slot)) = self.load_slow(ancestors, &account_pubkey) {
+                let frozen_account_info = FrozenAccountInfo {
+                    hash: Self::hash_frozen_account_data(&account),
+                    lamports: account.lamports,
+                };
+                warn!(
+                    "Account {} is now frozen at lamports={}, hash={}",
+                    account_pubkey, frozen_account_info.lamports, frozen_account_info.hash
+                );
+                self.frozen_accounts
+                    .insert(*account_pubkey, frozen_account_info);
+            } else {
+                panic!(
+                    "Unable to freeze an account that does not exist: {}",
+                    account_pubkey
+                );
+            }
+        }
+    }
+
     /// Store the account update.
     pub fn store(&self, slot_id: Slot, accounts: &[(&Pubkey, &Account)]) {
+        if !self.frozen_accounts.is_empty() {
```

Comment:
Can we put this block in a helper function, like check_frozen_accounts or process_frozen_accounts.

Reply:
Sure, I added `assert_frozen_accounts()`